public/main.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+let subReddits = [`https://www.reddit.com/r/Unity3D.json`, `https://www.reddit.com/r/blender.json`,
+`https://www.reddit.com/r/javascript.json`, `https://www.reddit.com/r/CivVI.json`,
+`https://www.reddit.com/r/Battletechgame.json`, `https://www.reddit.com/r/Cooking.json`, 
+`https://www.reddit.com/r/space.json`, `https://www.reddit.com/r/Guitar.json`, 
+`https://www.reddit.com/r/FullmetalAlchemist.json`, `https://www.reddit.com/r/Sculpture.json`,  ];
+let listingsArr = [];
+const postsElem =document.querySelector("#posts");
+const favButtonNames = ['#fav-button-one', '#fav-button-two', '#fav-button-three'];
+
+favButtonNames.forEach(function(name, index){
+  document.querySelector(name).addEventListener('click', () => requestListings(subReddits[index]));
+});
+document.querySelector('#random-button').addEventListener('click', function(){
+  const num = Math.floor(Math.random() * (subReddits.length - 3) + 3);
+  requestListings(subReddits[num]);
+});
+requestListings(subReddits[0]);
+
+function requestListings(subRedditURL){
+  postsElem.innerHTML = "";
+  const redditReq = new XMLHttpRequest();
+  redditReq.addEventListener('load', loadListings);
+  redditReq.open('GET', subRedditURL);
+  redditReq.send();
+}
+
+function loadListings(){
+  const listingsObj = JSON.parse(this.responseText);
+  listingsArr = listingsObj.data.children;
+
+  listingsArr.forEach(function(data){
+    const sectionNames = ['image', 'title', 'subtitle', 'text-snippet'];
+    const container = document.createElement('div');
+    postsElem.appendChild(container);
+    container.className = 'post';
+    
+    sectionNames.forEach(function(name){
```

Comment:
I don't think this forEach really help anything. Rather it makes it harder to read what's going on in the code. It might be let code to just move out the body if each if into the `listingArr.forEach`.